### PR TITLE
Add an opt-out label to runner determinator on PR

### PR DESCRIPTION
**Pull request by huydhn**
My sales pitch:  I need to ssh into the runner from time to time on my PR to debug issues, but it's well-known that LF runners don't support SSH login anymore.  So, the propose fix here is to introduce a new label called ~no-runner-determinator~ `no-runner-experiments` that can be attached to the PR.  Whenever `.github/scripts/runner_determinator.py` runs on a PR and sees this label, it will not apply any logic and just straight up use an empty prefix.

### Testing

With the label:

```
python3 runner_determinator.py \
    --github-token "MY_TOKEN" \
    --github-issue "5132" \
    --github-branch "install-torchao-torchtune-et" \
    --github-actor "huydhn" \
    --github-issue-owner "huydhn" \
    --github-ref-type "branch" \
    --github-repo "pytorch/pytorch" \
    --eligible-experiments "" \
    --pr-number "139947"

INFO    : Opt-out runner determinator because #139947 has no-runner-determinator label
WARNING : No env var found for GITHUB_OUTPUT, you must be running this code locally. Falling back to the deprecated print method.
::set-output name=label-type::
```

Without the label:

```
python3 runner_determinator.py \
    --github-token "MY_TOKEN" \
    --github-issue "5132" \
    --github-branch "install-torchao-torchtune-et" \
    --github-actor "huydhn" \
    --github-issue-owner "huydhn" \
    --github-ref-type "branch" \
    --github-repo "pytorch/pytorch" \
    --eligible-experiments "" \
    --pr-number "139947"

INFO    : Based on rollout percentage of 95%, enabling experiment lf.
INFO    : Skipping experiment 'awsa100', as it is not a default experiment
WARNING : No env var found for GITHUB_OUTPUT, you must be running this code locally. Falling back to the deprecated print method.
::set-output name=label-type::lf.
```

Running in trunk commit without a PR number will use the regular logic:

```
python3 runner_determinator.py \
    --github-token "MY_TOKEN" \
    --github-issue "5132" \
    --github-branch "install-torchao-torchtune-et" \
    --github-actor "huydhn" \
    --github-issue-owner "huydhn" \
    --github-ref-type "branch" \
    --github-repo "pytorch/pytorch" \
    --eligible-experiments "" \
    --pr-number ""

INFO    : Based on rollout percentage of 95%, enabling experiment lf.
INFO    : Skipping experiment 'awsa100', as it is not a default experiment
WARNING : No env var found for GITHUB_OUTPUT, you must be running this code locally. Falling back to the deprecated print method.
::set-output name=label-type::lf.
```

### CODE REVIEW

**Review by malfet — APPROVED** [review]
LGTM

File: .github/scripts/runner_determinator.py [hunk 1]
```diff
@@ -72,7 +76,7 @@
 GITHUB_OUTPUT = os.getenv("GITHUB_OUTPUT", "")
 GH_OUTPUT_KEY_AMI = "runner-ami"
 GH_OUTPUT_KEY_LABEL_TYPE = "label-type"
-
+OPT_OUT_LABEL = "no-runner-determinator"
```
Comment (ZainRizvi) [inline .github/scripts/runner_determinator.py:79]: nit: If we're disabling all the experiments, then let's label it accordingly

```suggestion
OPT_OUT_LABEL = "no-experiments"
```

File: .github/workflows/_runner-determinator.yml [hunk 2]
```diff
@@ -51,6 +51,7 @@ jobs:
       TRIGGERING_ACTOR: ${{ inputs.triggering_actor }}
       ISSUE_OWNER: ${{ inputs.issue_owner }}
       CHECK_EXPERIMENTS: ${{ inputs.check_experiments }}
+      PR_NUMBER: ${{ github.event.pull_request.number }}
```
Comment (ZainRizvi) [inline .github/workflows/_runner-determinator.yml:54]: Will this work for trunk-like jobs that are triggered by a tag being added?

**Review by ZainRizvi — APPROVED** [review]
This feels like a bit of a big hammer, since it disables all experiments rather than specifying the experiment that should be disabled.

That said, it's also a useful hammer to have on occasion and easier to use than modifying the issue that tracks every experiment's status.

Remaining concerns called out in the comments

File: .github/workflows/_runner-determinator.yml (hunk 2 above)
Comment (huydhn) [reply .github/workflows/_runner-determinator.yml:54]: Yeah, the code queries the labels from PR whenever it run, so the list is always fresh at the time `runnner_determinator.py` runs.  I basically copy this logic from `filter_test_config.py` script that I wrote a while back.  We know that the approach works

**Comment by huydhn** [comment]
Test run to confirm that Meta runner is picked up https://github.com/pytorch/pytorch/actions/runs/11732119060/job/32683896387#step:2:6

**Pull request closed without merging**